org.eclipse.sisu.plexus/src/main/java/org/codehaus/plexus/component/configurator/converters/basic/TemporalConverter.java

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.codehaus.plexus.component.configurator.converters.basic;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.Temporal;
+import java.time.temporal.TemporalAccessor;
+import java.util.Locale;
+
+import org.codehaus.plexus.component.configurator.ComponentConfigurationException;
+
+/**
+ * Supports type conversion into {@link java.time} classes.
+ * The supported patterns of the used {@link DateTimeFormatter} is either
+ * <ul>
+ * <li>ISO-8601 extended offset date-time-format ({@link DateTimeFormatter#ISO_OFFSET_DATE_TIME}) or</li>
+ * <li>{@code yyyy-MM-dd HH:mm:ss[[a][.S [a]]}</li>
+ * </ul>
+ *
+ */
+public class TemporalConverter
```

Comment:
@mcculls Any reason why `PlexusDateTypeConverter` was implemented as Guice `TypeConverter` and `DateConverter` is just an adapter?